core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -139,30 +140,15 @@ private synchronized final void closeWithTragicEvent(Throwable throwable) throws
         return new Translog.Location(generation, offset, data.length());
     }
 
+    private final ReentrantLock syncLock = new ReentrantLock();
```

Comment:
Maybe add a lock order comment here, e.g. "lock order syncLock -> synchronized(this)"?